In [2]:
import os
import sys
import numpy as np
import pylab as pl
import json
import os
%pylab inline

Populating the interactive namespace from numpy and matplotlib


NULL HYPOTHESIS: the $\%$ of former prisoners employed after release is the same or lower for candidates who participated in the program as for the control group, significance level $p=0.05$

$$ H_0: P_0−P_1 \geq 0$$
$$ H_a: P_0−P_1< 0$$
$$ \alpha = 0.05$$

This is a TEST OF PROPORTIONS. we use the Binomial distribution since it is a yes/no (bernulli) test for each subject: the former inmate was or was not ever employed in a CEO transitional job (second row in the table above): $P_0=0.035$ ,$P_1=0.701$

In [3]:
alpha=0.05
#we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0=3.5*0.01 
P_1=70.1*0.01

n_0=409
n_1=564

#lets get the counts by multiplying by the sample size
Nt_0=P_0*n_0
Nt_1=P_1*n_1

## Z TEST
the z test compares the stanrard deviation of the expected distribution and the observed result. it tells you literally how many standard deviations from the tail an observation is, under the _assumption of normality

In [4]:
#define the sample proportion first
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
print sp

0.421047276465


In [5]:
# i am goonna create a little one line function to calculate the standard dev, it is not really needed, but just to show you how you do such a thing
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )


sp_stdev_2y=sp_stdev((Nt_0+Nt_1)/(n_0+n_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_2y

0.035 409 564 0.0320658086057


In [6]:
#Print Z-score.
zscore = lambda p0, p1, s : (p0-p1)/s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print (z_2y)

20.7697865408


if $p< \alpha$ : reject $H_0$


IMPORTANT!! note that this $P$ is not the p-value, but
$p-$value$=1-P$

In [7]:
##p-value for employment after 2 years: since the largest number we read off the table for alpha=0.05 is smaller then the value for our statistic our p-value will be smaller then calculated using .9998
##thus using 0.9998 is a **conservative** approach. 

p_2y=1-0.9998


def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_2y,alpha)

is the p value 0.00 smaller than the critical value 0.05? 
YES!
the Null hypothesis is rejected


# Assignment:

NULL HYPOTHESIS: the $\%$ of former prisoners employed after release is the same or lower for candidates who participated in the program as for the control group, significance level $p=0.05$

$$ H_0: P_0−P_1 \geq 0$$
$$ H_a: P_0−P_1< 0$$
$$ \alpha = 0.05$$

##Here, I run the z test with the proportion of former prisioners ever employed in an unsubsidized job:
$P_0=0.521$ ,$P_1=0.533$

In [10]:
alpha=0.05
#we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0=52.1*0.01 
P_1=53.3*0.01

#lets get the counts by multiplying by the sample size
Nt_0=P_0*n_0
Nt_1=P_1*n_1
#Sample Proportion
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
print sp

0.527955806783


In [14]:
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )
sp_stdev_3y=sp_stdev((Nt_0+Nt_1)/(n_0+n_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_3y

 0.521 409 564 0.0324224126866


In [17]:
#Print Z-score.
zscore = lambda p0, p1, s : (p0-p1)/s
z_3y = zscore(P_1, P_0, sp_stdev_3y)
print (z_3y)   #value to look on the table

0.370114343926


So we look at the z test table for the value $0.37$ at the level of confidence $\alpha = 0.05$ ... 

In [20]:
p_3y=1-0.6443 #intersection between 0.3 and 0.07
 
report_result(p_3y,alpha)

is the p value 0.36 smaller than the critical value 0.05? 
NO!
the Null hypothesis is not rejected


## Now, with the $\chi ^2$ test 

|Employed for six or more consecutive quarters|Employed|Not Employed|
|------|------|------|------|
|Test Sample  |14.7 * 5.64  | 85.3 * 5.64  | 564 |
|Control Sample  |11.9 * 4.09  | 88.1 * 4.09  | 409 |
|Total  |131.58  | 841.42  | 973 |

In [23]:
Ntot = 973
expected = 564*409*131.58*841.42
sample_values = [[14.7*5.64,85.3*5.64],[11.9*4.09,88.1*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expect)

print chisqstat(Ntot,  sample_values, expected)

1.58938314156


## Questions

432 is hella larger then 3.84

why am i mentioning 3.84?

how does the 6mo chi square statistic that you derived compare?

please state what that means in terms of your hypothesis in a markdown cell below!

In the $ \chi ^2$ table, the columns represent our level  of tolerance (in this case $\alpha = 0.05$) for the Hypothesis and the rows represent the degrees of freedom of our test.

According to wikipedia (https://en.wikipedia.org/wiki/Chi-squared_test) the degrees of freedom are calculated by the equation: 

$$df= (\text{Number of rows} - 1 ) ( \text{Number of columns} - 1)$$ 

Well, that is the same as taking one from the number of categories that we are comparing (in this case we have just two categories: program and control group). In this case, we have just $1$ degree of freedom and $\alpha = 0.05$, then the value that we must look at the table is $3.84$.

On the example case, we got a  $\chi ^ 2$ result for "CEO transitional job" of approximately $432$ which is a number way greater than  $3.84$ on the table. Son in that case, the null hypothesis is rejected (therefore, the program group have a different performance than the control group).

In this notebook, I performed a test for the field "Employed for six or more consecutive quarters". The $\chi ^ 2$ value is: $1.589 < 3.84$. Furthermore, for one degree of freedom, the closest to  $1.589$ is  $1.32$ in the table, which corresponds to p = 0.25. Then, the null hypothesis of: 'the control group having a similar performance compared to the program group in the case of "Employment for six or more consecutive quarters"' cannot be rejected for the confidence level of $5 \% $

This result was expected, because the percentual difference between the categories in this second case was very small compared to the one in the example. 